frontend/lib/resolution/Resolver.cpp

```diff
@@ -1607,10 +1607,14 @@ void Resolver::adjustTypesForSplitInit(ID id,
   }
 
   const Param* p = rhsType.param();
-  if (lhsType.kind() != QualifiedType::PARAM) {
-    p = nullptr;
-  }
-  auto useType = QualifiedType(lhsType.kind(), rhsType.type(), p);
+  auto useKind = lhsType.kind();
+  if (p && symbol->isModule()) {
+    // White lie to let us save the param value for later, since at module level
+    // we may later learn the thing being split-init'd is param and needs it.
```

Comment:
> Is it if we resolve a module statement split-initing a previous module statement? We would have UnknownType in the _current_ resolver, which isn't param?

Yes, exactly. When we are in a Resolver for a particular module statement, the `ResolvedExpression` for other module statements seems to always just have the placeholder `UnknownType` since it's "out of scope". I'll add to the comment to explain this situation.